chia/full_node/full_node.py

```diff
@@ -1763,7 +1763,13 @@ async def respond_unfinished_block(
             await self.server.send_to_all_except([msg], NodeType.FULL_NODE, peer.peer_node_id)
         else:
             await self.server.send_to_all([msg], NodeType.FULL_NODE)
-        self._state_changed("unfinished_block")
+
+        state_changed_data = {}
```

Comment:
You might want to add more data here like:
- size of the generator bytes
- transaction generator ref list
- timestamp
- k size

Reply:
Or would that go into the new block state change?

Reply:
Moved all of this to `block` event. Didn't see timestamp immediately on `class FullBlock` - whats the best source for the timestamp? Where do I pull k size from in `respond_block`?

Reply:
It's all in the block itself. Check out the method `respond_block`, that's responsible for adding a block to the chain. After adding it, we can update the state for metrics